configs/opencv-ovms/cmd_client/main.go

```diff
@@ -342,7 +352,110 @@ func (ovmsClientConf *OvmsClientConfig) startOvmsServer() {
 
 	log.Println("Let server settle a bit...")
 	time.Sleep(ovmsSrvWaitTime)
-	log.Println("OVMS server started")
+
+	// wait for the model ready state status:
+	// even though the ovms server is ready but the individual models may not be
+	// due to the model config file config.json changes on the fly
+	retryCnt := 0
+	for {
+		if retryCnt >= ovmsModelReadyMaxRetries {
+			log.Printf("error: reaches the max retry count %d for checking model ready of ovms server; gave up", ovmsModelReadyMaxRetries)
+			return
+		}
+
+		retryCnt++
+		readyErr := ovmsClientConf.waitForOvmsModelsReady()
+		if readyErr == nil {
+			// all are error free and thus models are ready
+			break
+		}
+
+		log.Printf("warning: ovms models from the model server not ready yet: %v", readyErr)
+		// sleep a bit and retry it again
+		time.Sleep(time.Second)
+	}
+
+	log.Println("OVMS server started and ready to serve")
+}
+
+func (ovmsClientConf *OvmsClientConfig) waitForOvmsModelsReady() error {
+	grpcPort := os.Getenv(GRPC_PORT_ENV)
+	ovmsURL := fmt.Sprintf("%s:%s", "localhost", grpcPort)
```

Comment:
Moving the "localhost" to a variable will enable different hosts. The main case being a remote OVMS server.

Reply:
@brian-intel    let's deal with this in another enhancement PR as of today we are running docker in host mode network so it won't be able to support the remote yet